### DISCUSSION TITLE
Yourls instance is working sporadically

### DISCUSSION THREAD

**Discussion by arladmin** [YOURLS/YOURLS#3521] [Q&A]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/YOURLS/.github/blob/main/CODE_OF_CONDUCT.md).

### Submission validity

- [X] This is not a personal support request, that should be posted on [discussions community](https://github.com/YOURLS/YOURLS/discussions).
- [X] I checked [current issues](https://github.com/YOURLS/YOURLS/issues) and this request isn't a duplicate of an existing issue, opened or closed.

### Self troubleshooting

- [X] I've read the [documentation](https://yourls.org/) and made sure to follow it.
- [X] I've read the [troubleshooting first steps](https://docs.yourls.org/guide/troubleshooting/first-steps.html) and [frequent issues](https://docs.yourls.org/guide/troubleshooting/frequent-issues.html) guides.

### Version

1.9.1

### Description

I'm able to access the Admin page from Chrome, but everything else is unable to access the admin page.

```
Feb 14 09:02:30 - - - [14/Feb/2023:03:32:30 +0000] "GET /admin HTTP/1.1" 301 315 "-" "Mozilla (CloudronHealth)"
Feb 14 09:02:40 - - - [14/Feb/2023:03:32:40 +0000] "GET /admin HTTP/1.1" 301 315 "-" "Mozilla (CloudronHealth)"
Feb 14 09:02:50 - - - [14/Feb/2023:03:32:50 +0000] "GET /admin HTTP/1.1" 301 315 "-" "Mozilla (CloudronHealth)"
Feb 14 09:03:00 - - - [14/Feb/2023:03:33:00 +0000] "GET /admin HTTP/1.1" 301 315 "-" "Mozilla (CloudronHealth)"
Feb 14 09:03:00 172.x.x.x - - [14/Feb/2023:03:33:00 +0000] "GET /admin HTTP/1.1" 301 315 "-" "Uptime-Kuma/1.19.6"
Feb 14 09:03:10 - - - [14/Feb/2023:03:33:10 +0000] "GET /admin HTTP/1.1" 301 315 "-" "Mozilla (CloudronHealth)"
Feb 14 09:03:10 49.x.x.x - - [14/Feb/2023:03:33:10 +0000] "GET /admin HTTP/1.1" 301 315 "-" "PostmanRuntime/7.30.1"
Feb 14 09:03:20 - - - [14/Feb/2023:03:33:20 +0000] "GET /admin HTTP/1.1" 301 315 "-" "Mozilla (CloudronHealth)"
Feb 14 09:03:20 49.x.x.x - - [14/Feb/2023:03:33:20 +0000] "GET /admin/ HTTP/1.1" 200 5572 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
Feb 14 09:03:21 49.x.x.x - - [14/Feb/2023:03:33:21 +0000] "GET /js/jquery-3.5.1.min.js?v=1.9.1 HTTP/1.1" 200 30910 "https://my-domain/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
Feb 14 09:03:21 49.x.x.x - - [14/Feb/2023:03:33:21 +0000] "GET /js/common.js?v=1.9.1 HTTP/1.1" 200 2226 "https://my-domain/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
Feb 14 09:03:21 49.x.x.x - - [14/Feb/2023:03:33:21 +0000] "GET /js/jquery.notifybar.js?v=1.9.1 HTTP/1.1" 200 948 "https://my-domain/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
Feb 14 09:03:21 49.x.x.x - - [14/Feb/2023:03:33:21 +0000] "GET /css/style.css?v=1.9.1 HTTP/1.1" 200 1901 "https://my-domain/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
Feb 14 09:03:21 49.x.x.x - - [14/Feb/2023:03:33:21 +0000] "GET /css/tablesorter.css?v=1.9.1 HTTP/1.1" 200 766 "https://my-domain/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
Feb 14 09:03:21 49.x.x.x - - [14/Feb/2023:03:33:21 +0000] "GET /js/jquery-3.tablesorter.min.js?v=1.9.1 HTTP/1.1" 200 14298 "https://my-domain/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
Feb 14 09:03:21 49.x.x.x - - [14/Feb/2023:03:33:21 +0000] "GET /js/tablesorte.js?v=1.9.1 HTTP/1.1" 200 753 "https://my-domain/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
Feb 14 09:03:21 49.x.x.x - - [14/Feb/2023:03:33:21 +0000] "GET /js/insert.js?v=1.9.1 HTTP/1.1" 200 2273 "https://my-domain/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
Feb 14 09:03:21 49.x.x.x - - [14/Feb/2023:03:33:21 +0000] "GET /css/share.css?v=1.9.1 HTTP/1.1" 200 480 "https://my-domain/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
Feb 14 09:03:21 49.x.x.x - - [14/Feb/2023:03:33:21 +0000] "GET /js/share.js?v=1.9.1 HTTP/1.1" 200 665 "https://my-domain/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
Feb 14 09:03:21 49.x.x.x - - [14/Feb/2023:03:33:21 +0000] "GET /js/clipboard.min.js?v=1.9.1 HTTP/1.1" 200 3354 "https://my-domain/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
Feb 14 09:03:21 49.x.x.x - - [14/Feb/2023:03:33:21 +0000] "GET /css/cal.css?v=1.9.1 HTTP/1.1" 200 404 "https://my-domain/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
Feb 14 09:03:21 49.x.x.x - - [14/Feb/2023:03:33:21 +0000] "GET /js/jquery.cal.js?v=1.9.1 HTTP/1.1" 200 3919 "https://my-domain/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
Feb 14 09:03:21 49.x.x.x - - [14/Feb/2023:03:33:21 +0000] "GET /images/yourls-logo.svg?v=1.9.1 HTTP/1.1" 200 14061 "https://my-domain/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
Feb 14 09:03:21 49.x.x.x - - [14/Feb/2023:03:33:21 +0000] "GET /images/chart_bar.svg HTTP/1.1" 200 1708 "https://my-domain/css/style.css?v=1.9.1" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
Feb 14 09:03:21 49.x.x.x - - [14/Feb/2023:03:33:21 +0000] "GET /images/share.svg HTTP/1.1" 200 2005 "https://my-domain/css/style.css?v=1.9.1" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
Feb 14 09:03:21 49.x.x.x - - [14/Feb/2023:03:33:21 +0000] "GET /images/pencil.svg HTTP/1.1" 200 1796 "https://my-domain/css/style.css?v=1.9.1" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
Feb 14 09:03:21 49.x.x.x - - [14/Feb/2023:03:33:21 +0000] "GET /images/delete.svg HTTP/1.1" 200 1883 "https://my-domain/css/style.css?v=1.9.1" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
Feb 14 09:03:21 49.x.x.x - - [14/Feb/2023:03:33:21 +0000] "GET /images/favicon.svg HTTP/1.1" 200 6558 "https://my-domain/css/style.css?v=1.9.1" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
Feb 14 09:03:22 49.x.x.x - - [14/Feb/2023:03:33:22 +0000] "GET /images/bg.svg HTTP/1.1" 200 1762 "https://my-domain/css/tablesorter.css?v=1.9.1" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
Feb 14 09:03:22 49.x.x.x - - [14/Feb/2023:03:33:22 +0000] "GET /images/desc.svg HTTP/1.1" 200 1702 "https://my-domain/css/tablesorter.css?v=1.9.1" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
Feb 14 09:03:30 - - - [14/Feb/2023:03:33:30 +0000] "GET /admin HTTP/1.1" 301 315 "-" "Mozilla (CloudronHealth)"
Feb 14 09:03:40 - - - [14/Feb/2023:03:33:40 +0000] "GET /admin HTTP/1.1" 301 315 "-" "Mozilla (CloudronHealth)"
```

Can't figure out what's wrong in this case.

### Context

My .htaccess 

```
# BEGIN YOURLS
<IfModule mod_rewrite.c>
RewriteEngine On
RewriteBase /
RewriteCond %{HTTPS} !=on
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule ^.*$ /yourls-loader.php [L]
</IfModule>
# END YOURLS
```

**Comment by ozh** [comment]
> I'm able to access the Admin page from Chrome, but everything else is unable to access the admin page.

I don't understand this phrase. What works "sporadically" ? (which means "sometimes it works, sometimes it doesn't")

**Reply by arladmin** [reply]
As you can see from the log, some requests are 200, then some are not, and repeat. That's why i mentioned sporadically.

**Comment by logusivam** [comment]
The logs tell the full story here. Your `.htaccess` has the same issue as a common misplaced-condition pattern, but with a specific twist visible in your logs that explains why Chrome works and everything else loops.

---

## Reading the logs — what's actually happening

```
"GET /admin HTTP/1.1" 301  ← CloudronHealth, Uptime-Kuma, Postman (first request)
"GET /admin/ HTTP/1.1" 200 ← Chrome (after following the 301)
```

There are **two separate redirects** chained here:

1. **HTTP → HTTPS redirect** (your `RewriteCond %{HTTPS} !=on` rule) — firing for any non-HTTPS request
2. **Trailing slash redirect** — `/admin` → `/admin/` (Apache's `DirectorySlash` or YOURLS's own handling)

Chrome follows both redirects silently and lands on `/admin/` over HTTPS — so it works. Health checkers like CloudronHealth send plain HTTP requests (`"Mozilla (CloudronHealth)"` with no HTTPS), hit redirect #1, and depending on their configuration either don't follow the 301 at all, or follow it and hit another condition that sends them back — resulting in the repeated 301s you see every 10 seconds.

---

## The root cause — same misplaced `RewriteCond` as a classic pattern

```apache
RewriteCond %{HTTPS} !=on          ← Problem: this gates the YOURLS rule
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule ^.*$ /yourls-loader.php [L]
```

All three `RewriteCond` lines apply to the **single** `RewriteRule` below them — Apache ANDs all conditions together. This means YOURLS's routing rule only fires when:

- The connection is **not** HTTPS, **AND**
- The file doesn't exist, **AND**
- The directory doesn't exist

So HTTPS visitors (Chrome) bypass the YOURLS routing rule entirely and instead get served by Apache's default directory handling — which happens to serve `/admin/` correctly because the directory physically exists. Non-HTTPS clients (health checkers) hit the YOURLS rule, get sent to `yourls-loader.php`, which then issues its own redirect... creating the loop.

The fact that Chrome works is a coincidence of Apache's `DirectorySlash` behavior serving the physical `/admin/` directory directly — not because your rewrite rules are correct.

---

## The fix

Separate the HTTP→HTTPS redirect into its own rule block, completely independent of the YOURLS routing rule:

```apache
# BEGIN YOURLS
<IfModule mod_rewrite.c>
RewriteEngine On
RewriteBase /

# Rule 1: Force HTTP → HTTPS (self-contained rule, separate from YOURLS routing)
RewriteCond %{HTTPS} !=on
RewriteRule ^ https://%{HTTP_HOST}%{REQUEST_URI} [L,R=301]

# Rule 2: YOURLS routing — no HTTPS condition here
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule ^.*$ /yourls-loader.php [L]
</IfModule>
# END YOURLS
```

With this structure:
- HTTP requests hit Rule 1 and get permanently redirected to HTTPS — **one** clean 301, no loop
- HTTPS requests skip Rule 1 (condition is false), fall through to Rule 2, and YOURLS handles the routing correctly
- Health checkers sending HTTP will get a single 301 to HTTPS, then a 200 on the follow-up — or if they're configured to monitor HTTPS directly, they'll hit Rule 2 immediately and get a 200

---

## If your health checkers must monitor over plain HTTP (no HTTPS follow)

Some monitoring tools (especially self-hosted ones like Uptime-Kuma in certain configurations) don't follow redirects and expect a direct 200. If that's the case, configure them to monitor the **HTTPS URL** directly:

```
https://yourdomain.com/admin/
```

rather than:

```
http://yourdomain.com/admin
```

This way they never hit the HTTP→HTTPS redirect at all. The trailing slash on `/admin/` also matters — monitor `/admin/` not `/admin` to avoid the directory-slash redirect as a second hop.

---

## Why Chrome worked despite the broken config

Chrome follows redirect chains transparently and handles both the HTTP→HTTPS and the `/admin` → `/admin/` trailing-slash redirect in a single seamless sequence. It also caches the 301 permanently, so after the first visit it goes straight to `https://yourdomain.com/admin/` without hitting the redirect chain again — which is why you may not have noticed anything was wrong on the browser side.

Health checkers don't have that luxury — they start fresh every check interval, and many don't follow redirects or have a maximum redirect limit of 0 or 1, which is why the misconfiguration is visible only in their logs.

---

## Summary

| Client | What was happening | After fix |
|--------|-------------------|-----------|
| Chrome (HTTPS) | Accidentally worked via Apache directory serving, not YOURLS | Works correctly via YOURLS routing |
| CloudronHealth (HTTP) | Stuck in 301 loop every 10 seconds | Single 301 to HTTPS, then 200 |
| Uptime-Kuma (HTTP) | Same 301 loop | Configure to monitor HTTPS URL directly |
| Postman (HTTP) | Got 301 then 200 after following redirect | Works correctly |

Hope this helps — if it answers your question, would you mind closing this issue or clicking **"Mark as answer"** so others searching for the same thing can find it quickly?
